includes/Checker/Preparations/Use_Minimal_Theme_Preparation.php

```diff
@@ -0,0 +1,142 @@
+/**
+ * Class for the preparation step to force usage of a minimal theme.
+ *
+ * This ensures the plugin is checked as much in isolation as possible.
+ *
+ * @since n.e.x.t
+ */
+class Use_Minimal_Theme_Preparation implements Preparation {
+
+	/**
+	 * Theme slug / directory name.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	protected $theme_slug;
+
+	/**
+	 * Absolute path to themes root directory.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	protected $themes_dir;
+
+	/**
+	 * Sets the theme slug and themes root directory.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $theme_slug Slug of the theme to enforce.
+	 * @param string $themes_dir Optional. Absolute path to themes root directory, if not the regular wp-content/themes.
+	 */
+	public function __construct( $theme_slug, $themes_dir = '' ) {
+		$this->theme_slug = $theme_slug;
+		$this->themes_dir = $themes_dir;
+	}
+
+	/**
+	 * Runs this preparation step for the environment and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		// Override the theme slug and name.
+		add_filter( 'template', array( $this, 'get_theme_slug' ) );
+		add_filter( 'stylesheet', array( $this, 'get_theme_slug' ) );
+		add_filter( 'pre_option_template', array( $this, 'get_theme_slug' ) );
+		add_filter( 'pre_option_stylesheet', array( $this, 'get_theme_slug' ) );
+		add_filter( 'pre_option_current_theme', array( $this, 'get_theme_name' ) );
+
+		// Override the theme directory.
+		add_filter( 'pre_option_template_root', array( $this, 'get_theme_root' ) );
+		add_filter( 'pre_option_stylesheet_root', array( $this, 'get_theme_root' ) );
+
+		// Register the custom themes directory if relevant.
+		if ( ! empty( $this->themes_dir ) ) {
+			register_theme_directory( $this->themes_dir );
+
+			// Force new directory scan to ensure the new theme directory is available.
+			search_theme_directories( true );
+		}
+
+		// Return the cleanup function.
+		return function() {
+			global $wp_theme_directories;
+
+			remove_filter( 'template', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'stylesheet', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'pre_option_template', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'pre_option_stylesheet', array( $this, 'get_theme_slug' ) );
+			remove_filter( 'pre_option_current_theme', array( $this, 'get_theme_name' ) );
+
+			remove_filter( 'pre_option_template_root', array( $this, 'get_theme_root' ) );
+			remove_filter( 'pre_option_stylesheet_root', array( $this, 'get_theme_root' ) );
+
+			if ( ! empty( $this->themes_dir ) ) {
+				$index = array_search( untrailingslashit( $this->themes_dir ), $wp_theme_directories, true );
+				if ( false !== $index ) {
+					array_splice( $wp_theme_directories, $index, 1 );
+					$wp_theme_directories = array_values( $wp_theme_directories );
```

Comment:
Since you fixed the transient problem above via `search_theme_directories( true )` call (good catch!), you probably need to do the same here, otherwise the list of themes will continue to include themes from our custom directory even later.